### PR TITLE
Impl/copilot toggle

### PR DESCRIPTION
**Pull request by TerminalFi**

### CODE REVIEW

File: plugin/ui/chat.py [hunk 1]
```diff
@@ -117,8 +126,11 @@ def conversation(self, value: list[CopilotPayloadConversationEntry]) -> None:
     # normal methods #
     # -------------- #
 
-    def __init__(self, window: sublime.Window) -> None:
-        self.window = window
+    def __init__(self, window: sublime.Window | None) -> None:
+        if not window:
+            raise ValueError("window is required")
+
+        self.window: sublime.Window = window
```
Comment (jfcherng) [inline plugin/ui/chat.py:133]: What's going on? If we need this check, then the issue is probably in other places.

File: plugin/ui/chat.py [hunk 2]
```diff
@@ -169,8 +181,8 @@ def close(self) -> None:
 
 class _ConversationEntry:
     def __init__(self, window: sublime.Window) -> None:
-        self.window = window
-        self.wcm = WindowConversationManager(window)
+        self.window: sublime.Window = window
+        self.wcm: WindowConversationManager = WindowConversationManager(window)
```
Comment (jfcherng) [inline plugin/ui/chat.py:185]: Tools should be able to auto recognize these types correctly?

File: plugin/utils.py [hunk 3]
```diff
@@ -111,15 +111,25 @@ def fix_completion_syntax_highlight(view: sublime.View, point: int, code: str) -
     return code
 
 
-def get_copilot_setting(instance: sublime.Window | sublime.View, prefix: str, key: str, default: Any = None) -> Any:
+def get_copilot_setting(
+    instance: sublime.Window | sublime.View | None, prefix: str, key: str, default: Any = None
```
Comment (jfcherng) [inline plugin/utils.py:115]: Allowing `None` smells bad imo...

File: plugin/ui/chat.py (hunk 1 above)
Comment (TerminalFi) [reply plugin/ui/chat.py:133]: Ci-check was complaining

File: plugin/utils.py (hunk 3 above)
Comment (TerminalFi) [reply plugin/utils.py:115]: I can remove. Ci-check was just complaining

**Comment by jfcherng** [comment]
I can't seem to reproduce those ci-check complaints so I just revised them back to what they were.

**Review by jfcherng — APPROVED** [review]

**Pull request merged**